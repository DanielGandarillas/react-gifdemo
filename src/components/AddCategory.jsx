import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value)
    console.log(target.value)
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    const value = inputValue.trim();
    if (value.length <= 0) return;
    onNewCategory(value)
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}