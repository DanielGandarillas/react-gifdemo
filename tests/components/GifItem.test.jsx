import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"



describe('Prueba de <GifItem />', () => {

    const title = "Soy un titulo";
    const url = "https://soyurl.test/";

    test('debe de validar el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe de mostar el titulo del componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})