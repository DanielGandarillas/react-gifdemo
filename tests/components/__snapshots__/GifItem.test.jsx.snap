// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifItem /> debe de validar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un titulo"
      src="https://soyurl.test/"
    />
    <p>
      Soy un titulo
    </p>
  </div>
</div>
`;
